CodeMaid/Helpers/CodeItemTypeComparer.cs

```diff
@@ -118,16 +119,20 @@ private static int CalculateAccessOffset(BaseCodeItem codeItem)
             var codeItemElement = codeItem as BaseCodeItemElement;
             if (codeItemElement == null) return 0;
 
-            switch (codeItemElement.Access)
-            {
-                case vsCMAccess.vsCMAccessPublic: return 1;
-                case vsCMAccess.vsCMAccessAssemblyOrFamily: return 2;
-                case vsCMAccess.vsCMAccessProject: return 3;
-                case vsCMAccess.vsCMAccessProjectOrProtected: return 4;
-                case vsCMAccess.vsCMAccessProtected: return 5;
-                case vsCMAccess.vsCMAccessPrivate: return 6;
-                default: return 0;
-            }
+            var itemsOrder = new[] {
+                vsCMAccess.vsCMAccessPublic,
+                vsCMAccess.vsCMAccessAssemblyOrFamily,
+                vsCMAccess.vsCMAccessProject,
+                vsCMAccess.vsCMAccessProjectOrProtected,
+                vsCMAccess.vsCMAccessProtected,
+                vsCMAccess.vsCMAccessPrivate
+                };
+            if (!Settings.Default.Reorganizing_RevertAccessLevel)
+                itemsOrder = itemsOrder.Reverse().ToArray();
+
+            return itemsOrder.Contains(codeItemElement.Access)
```

Comment:
IndexOf always returns -1 when no match is found, so you could remove the conditional and always return IndexOf+1.